File: easy/stepwise_word.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    next if ( $line =~ m/^\s$/ );
    chomp($line);
    my @s = split( / /, $line );
    my $r;
    map {
        if ( length $_ > length $r ) { $r = $_ }
    } @s;
    my @a = map { "*" x $_ . substr( $r, $_, 1 ) } 0 .. ( length $r ) - 1;
    printf( "%s\n", join( ' ', @a ) );
}
close(INFILE);

File: easy/stepwise_word.rb
File.open(ARGV[0]).each_line do |line|
  s = line.chomp.split
  m = ''
  s.each { |i| m = i if i.length > m.length }
  puts m.split('').each_with_index.map { |i, ix| '*' * ix + i }.join(' ')
end
